htdocs/compta/facture/list.php

```diff
@@ -442,22 +442,23 @@
 $sql .= " LEFT JOIN ".MAIN_DB_PREFIX."c_departements as state on (state.rowid = s.fk_departement)";
 if (!empty($search_categ_cus)) $sql .= ' LEFT JOIN '.MAIN_DB_PREFIX."categorie_societe as cc ON s.rowid = cc.fk_soc"; // We'll need this table joined to the select in order to filter by categ
 
-$sql .= ', '.MAIN_DB_PREFIX.'facture as f';
+$sql .= ' INNER JOIN '.MAIN_DB_PREFIX.'facture as f ON f.fk_soc = s.rowid'; // INNER JOIN and , (comma) are semantically equivalent in the absence of a join condition: both produce a Cartesian product between the specified tables (that is, each and every row in the first table is joined to each and every row in the second table).
+
 if (is_array($extrafields->attributes[$object->table_element]['label']) && count($extrafields->attributes[$object->table_element]['label'])) $sql .= " LEFT JOIN ".MAIN_DB_PREFIX.$object->table_element."_extrafields as ef on (f.rowid = ef.fk_object)";
 if (!$sall) $sql .= ' LEFT JOIN '.MAIN_DB_PREFIX.'paiement_facture as pf ON pf.fk_facture = f.rowid';
 if ($sall || $search_product_category > 0) $sql .= ' LEFT JOIN '.MAIN_DB_PREFIX.'facturedet as pd ON f.rowid=pd.fk_facture';
 if ($search_product_category > 0) $sql .= ' LEFT JOIN '.MAIN_DB_PREFIX.'categorie_product as cp ON cp.fk_product=pd.fk_product';
 $sql .= " LEFT JOIN ".MAIN_DB_PREFIX."projet as p ON p.rowid = f.fk_projet";
 $sql .= ' LEFT JOIN '.MAIN_DB_PREFIX.'user AS u ON f.fk_user_author = u.rowid';
 // We'll need this table joined to the select in order to filter by sale
-if ($search_sale > 0 || (!$user->rights->societe->client->voir && !$socid)) $sql .= ", ".MAIN_DB_PREFIX."societe_commerciaux as sc";
+if ($search_sale > 0 || (!$user->rights->societe->client->voir && !$socid)) $sql .= " INNER JOIN ".MAIN_DB_PREFIX."societe_commerciaux as sc ON s.rowid = sc.fk_soc"; // INNER JOIN and , (comma) are semantically equivalent in the absence of a join condition: both produce a Cartesian product between the specified tables (that is, each and every row in the first table is joined to each and every row in the second table).
 if ($search_user > 0)
 {
 	$sql .= ", ".MAIN_DB_PREFIX."element_contact as ec";
 	$sql .= ", ".MAIN_DB_PREFIX."c_type_contact as tc";
 }
 
-$sql .= ' WHERE f.fk_soc = s.rowid';
+$sql .= ' WHERE f.fk_soc = s.rowid'; // Shoul
```

Comment:
If you add an inner join with a on clause, you should not have the clause into the where.
Using , should be possible if "left join" are always positionned just after the from of table they are linked too.
Which sql error do you get ? Is it on posgresql ?

Reply:
Its on maria db: 

`B_ERROR_NOSUCHFIELD: Unknown column 'f.fk_user_author' in 'on clause'`

When adding the inner join before *facture as f* I've got an:

`B_ERROR_NOSUCHFIELD: Unknown column 'f sc.fk_soc' in 'on clause'`

I am not an sql expert, but I have already experienced a problem of the same type on another project. I think that the problem is more related to the cumulative alias (as) and join (left join) than to the order of the instructions, cause, as I remember, removing aliases un the other project has solved the problem. 

It'is not a clean way coding, but I wasn't aware of what could ne the aide effect of removing the condition from the where clause, so I didn't, but you're right they sould be removed.

Reply:
Can you confirm you have the error after pulling last version, without your PR ?

Reply:
I've pulled last version with a clean base (migrate from prod) and I dont have the probleme anymore.

I dont have error anymore.